invoice/src/main/java/org/killbill/billing/invoice/usage/RawUsageOptimizer.java

```diff
@@ -122,7 +122,9 @@ public BillingPeriod apply(final Usage input) {
             final InvoiceItem item = iterator.previous();
             if (!(item instanceof UsageInvoiceItem)) {
                 // Help to debug https://github.com/killbill/killbill/issues/1095
-                log.warn("RawUsageOptimizer : item id={}, expected to see an UsageInvoiceItem type, got {}", item.getId(), item);
+                log.warn("RawUsageOptimizer : item id={}, expected to see an UsageInvoiceItem type, got class {}",
+                         item.getId(),
+                         (item != null ? item.getClass() : null));
```

Comment:
Maybe you should also print the class loader?

Reply:
See https://github.com/killbill/killbill/pull/1247/commits/0139d1462046a1fbf4d3bc60fb0557cf21cdac51